### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by wadaniel**
Hi guys, this is the refactoring we discussed in December:

3) Do refactoring: (D+P+S)
	1) solver/agent -> solver/ReinforcementLearner
	3) solver/learner/deepSupervisor -> solver/SupervisedLearner
	4) solver/learner [DELETE]
	5) _agent -> _worker in agent.hpp and .cpp
	6) _stateVector -> _stateBuffer and for others too
	7) _sessionAgentComputationTime -> Worker
	8) attendAgent -> attendWorker
	9) getAgentPolicy -> getPolicy
	10) printAgentInformation -> printInformation
	11) Concurrent Environments -> concurrent workers

### CODE REVIEW

File: tests/unit/modules/solver/learners.cpp [hunk 1]
```diff
@@ -2,7 +2,6 @@
 #include "korali.hpp"
 #include "modules/solver/learner/learner.hpp"
 #include "modules/solver/learner/deepSupervisor/deepSupervisor.hpp"
```
[the remark this answers is not in the document]
Comment (wadaniel) [reply tests/unit/modules/solver/learners.cpp:4]: this is not yet done, thanks

File: source/modules/problem/supervisedLearning/supervisedLearning.config [hunk 2]
```diff
@@ -47,7 +47,7 @@
    }
  ],
 
- "Compatible Solvers": [ "Learner/DeepSupervisor", "Learner/Gaussian Process" ],
+ "Compatible Solvers": [ "Learner/DeepSupervisor" ],
```
[the remark this answers is not in the document]
Comment (wadaniel) [reply source/modules/problem/supervisedLearning/supervisedLearning.config:50]: this is not yet done, thanks

File: source/modules/problem/reinforcementLearning/reinforcementLearning.cpp [hunk 3]
```diff
@@ -199,9 +242,6 @@ void ReinforcementLearning::runTrainingEpisode(Sample &agent)
   message["Episodes"] = episodes;
   KORALI_SEND_MSG_TO_ENGINE(message);
 
-  // Finalizing Environment
```
[the remark this answers is not in the document]
Comment (wadaniel) [reply source/modules/problem/reinforcementLearning/reinforcementLearning.cpp:202]: taken from MARL: https://github.com/cselab/korali/blob/MARL/source/modules/problem/reinforcementLearning/reinforcementLearning.cpp.base

File: tests/unit/modules/solver/learners.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (wadaniel) [reply tests/unit/modules/solver/learners.cpp]: fixed

**Review by webepasc — APPROVED** [review]

**Review by webepasc — APPROVED** [review]

**Pull request merged**